sphinx/util/texescape.py

```diff
@@ -46,6 +46,14 @@
     ('|', r'\textbar{}'),
     ('ℯ', r'e'),
     ('ⅈ', r'i'),
+    # Greek alphabet not escaped: pdflatex handles it via textalpha and inputenc
+    # OHM SIGN U+2126 is handled by LaTeX textcomp package
+]
+
+# A map Unicode characters to LaTeX representation
+# (for LaTeX engines which don't support unicode)
+unicode_tex_replacements = [
```

Comment:
I don't know what characters should be kept as-is.

Reply:
I tested  the characters (one on each line below):

```
    ¶
    §
    €
    ∞
    ±
    →
    ‣
    ✓
    ✔
    ﻿
    ⎽
    –
    |
    ℯ
    ⅈ
```

with xelatex and GNU FreeFont. 

- Serif contains them all
- Sans and Mono are lacking some.

Here are the relevant warnings: (you need to see that with a font which has the glyphs...)

```
Missing character: There is no ✓ in font [FreeSans.otf]/OT:script=latn;language
=DFLT;mapping=tex-text;!
Missing character: There is no ✔ in font [FreeSans.otf]/OT:script=latn;language
=DFLT;mapping=tex-text;!
Missing character: There is no ﻿ in font [FreeSans.otf]/OT:script=latn;language
=DFLT;mapping=tex-text;!
Missing character: There is no ⎽ in font [FreeSans.otf]/OT:script=latn;language
=DFLT;mapping=tex-text;!
Missing character: There is no ⅈ in font [FreeSans.otf]/OT:script=latn;language
=DFLT;mapping=tex-text;!
 
Missing character: There is no ✓ in font [FreeMono.otf]/OT:script=latn;language
=DFLT;!
Missing character: There is no ✔ in font [FreeMono.otf]/OT:script=latn;language
=DFLT;!
Missing character: There is no ℯ in font [FreeMono.otf]/OT:script=latn;language
=DFLT;!
Missing character: There is no ⅈ in font [FreeMono.otf]/OT:script=latn;language
=DFLT;!
```

Sans is used in titles and Mono in code-blocks.

I thus think a daring but still cautious approach currently would be maintain the TeX escaping even with Unicode engines for the following characters: ``✓ , ✔ , ⎽ ,  ℯ , ⅈ`` and the ZWNBSP  (U+FEFF, also used as BYTE ORDER MARK).

But for ``¶,  §,  €,  ∞,  ±,  →,  ‣,  –,  |`` I don't see obvious reason why they should be TeX-escaped with Unicode engines.

Note 1: contrarily to web browsers, LaTeX has no easily usable concept of "fall-back" font when a glyph is missing from a font. It is up to user to add some `\DeclareUnicodeCharacter` or `\newunicodechar` to tell LaTeX that for this or that glyph, another font must be used. And making this fontfamily known to fontspec is also under user responsabillity.

Note 2: one may hesitate about escaping `ℯ` as it is availble in FreeSerif, but it is lacking from FreeMono and also from DejaVu Sans Mono. (as seen in #6733 original report). Probably same for `ⅈ`.

I am tempted to not escape them, but hope that user community will share advice on how to use `\newunicodechar` (I have not looked myself in detail and in particular how to use a fallback font for them only when needed, i.e. not in normal text which uses FreeSerif).